src/openloco/graphics/gfx.cpp

```diff
@@ -184,10 +184,82 @@ namespace openloco::gfx
      */
     uint16_t get_string_width(const char* buffer)
     {
-        registers regs;
-        regs.esi = (uintptr_t)buffer;
-        call(0x495685, regs);
-        return regs.cx;
+        uint16_t width = 0;
+        uint8_t* str = (uint8_t*)buffer;
+
+        while (*str != (uint8_t)0)
+        {
+            uint8_t chr = *str;
+            str++;
+
+            if (chr >= 32)
+            {
+                width += _characterWidths[chr - 32 + _currentFontSpriteBase];
+                continue;
+            }
+
+            switch (chr)
+            {
+                case control_codes::move_x:
+                    width = *str;
+                    str++;
+                    break;
+
+                case control_codes::adjust_palette:
+                case 3:
+                case 4:
+                    str++;
+                    break;
+
+                case control_codes::newline:
+                case control_codes::newline_smaller:
+                    continue;
```

Comment:
stick to using `break` over `continue` for this as they all follow the same codepath. This also applies below for the <= 16 statements.